src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle current = (Rectangle) obj;
+        return (width == null && current.width == null
+                    || (width != null && width.equals(current.width)))
+                && (length == null && current.length == null
+                    || (length != null && length.equals(current.length)))
+                && (color == null && current.color == null
+                    || (color != null && color.equals(current.color)));
```

Comment:
```suggestion
        return (width == current.width
                    || (width != null && width.equals(current.width)))
                && (length == current.length
                    || (length != null && length.equals(current.length)))
                && (color == current.color
                    || (color != null && color.equals(current.color)));
```
this can be simplified a bit

Reply:
:+1: